astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+                dtype = col.dtype.str
+                if col.has_null:
+                    mcol = col
+                    mcol.fill_value = ""
+                    coltmp = Column(mcol.filled(), dtype=str)
+                    dtype = coltmp.dtype.str
+                if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                    col.meta.size = int(re.sub(r'^[^0-9]+(\d+)$', r'\1', dtype))
+                col.fortran_format = "A" + str(col.meta.size)
+                col.format = str(col.meta.size) + "s"
+
+            endb = col.meta.size + startb - 1
+
+            # ``mixin`` columns converted to string valued columns will not have a name
+            # attribute. In those cases, a ``Unknown`` column label is put, indicating that
+            # such columns can be better formatted with some manipulation before calling
+            # the CDS/MRT writer.
+            if col.name is None:
+                col.name = "Unknown"
+
+            # Set column description.
+            if col.description is not None:
+                description = col.description
+            else:
+                description = "Description of " + col.name
+
+            # Set null flag in column description
+            nullflag = ""
+            if col.has_null:
+                nullflag = "?"
+
+            # Set column unit
+            if col.unit is not None:
+                col.meta.unit = col.unit.to_string("cds")
+            elif col.name.lower().find("magnitude") > -1:
+                # ``col.unit`` can still be ``None``, if the unit of column values
+                # is ``Magnitude``, because ``astropy.units.Magnitude`` is actually a class.
+                # Unlike other units which are instances of ``astropy.units.Unit``,
+                # application of the ``Magnitude`` unit calculates the logarithm
+                # of the values. Thus, the only way to check for if the column values
+                # have ``Magnitude`` unit is to check the column name.
+                col.meta.unit = "mag"
+            else:
+                col.meta.unit = "---"
+
+            # Add col limit values to col description
+            lim_vals = ""
+            if (col.min and col.max and
+                not any(x in col.name for x in ['RA', 'DE', 'LON', 'LAT'])):
+                # No col limit values for coordinate columns.
+                if col.fortran_format[0] == 'I':
+                    if abs(col.min) < MAX_COL_INTLIMIT and abs(col.max) < MAX_COL_INTLIMIT:
+                        if col.min == col.max:
+                            lim_vals = "[{0}]".format(col.min)
+                        else:
+                            lim_vals = "[{0}/{1}]".format(col.min, col.max)
+                elif col.fortran_format[0] in ('E', 'F'):
+                    lim_vals = "[{0}/{1}]".format(math.floor(col.min * 100) / 100.,
+                                                  math.ceil(col.max * 100) / 100.)
```

Comment:
I think the limit values are how the CDS/MRT services check if the values within the column designated by RAh, for instance, fall within the defined limits. Checkout the following given towards the end of section 3.4.1 in http://vizier.u-strasbg.fr/doc/catstd-3.4.htx. MRT's page also has the same (https://journals.aas.org/mrt-explanation/).

> When specified, limiting numbers should represent actual limits, and not the range of all possible values which can be inferred from the format (e.g. [-999,9999] for an I4 number).
> Some labels have implicit limits, listed in the column Limits of the tables in section 3.3. These defaults are overridden (for numeric columns only) by the limits specified within square brackets in the description file : writing e.g.[] as the first word of the explanation of a column labelled GLON removes the condition 0 <=GLON < 360.

Reply:
However, I do agree that the using `MAX_COL_INTLIMIT` and writing out max/min values inside `[...]` is not the best implementation. As you say, using a `limits` argument will be a better approach, if at all we do want to include them.
This would need some work. So, **needs an issue**.